toolchain/check/impl_lookup.cpp

```diff
@@ -145,14 +145,11 @@ auto LookupImplWitness(Context& context, SemIR::LocId loc_id,
     // constraint, the unique `interface` declaration.)
 
     auto specific_id = SemIR::SpecificId::None;
-    // This check comes first to avoid deduction when the witness_id is missing.
-    // We use an error value to indicate an error during creation of the impl,
-    // such as a recursive impl which will cause deduction to recurse
-    // infinitely.
-    if (!impl.witness_id.has_value() ||
-        impl.witness_id == SemIR::ErrorInst::SingletonInstId) {
-      // TODO: Diagnose if the impl isn't defined yet?
-      return SemIR::InstId::None;
```

Comment:
`return` here was preventing us from finding a matching impl if there was an error in an earlier impl.